### DISCUSSION TITLE
Keyboard shortcut commands to resize *OS* window?

### DISCUSSION THREAD

**Discussion by egnor** [kovidgoyal/kitty#3839] [Ideas]
These wouldn't need to be in any default keymapping, but it would be nice to have commands that could set the width and/or height to a specific value (e.g. 80c), a specific offset (e.g. +40c), or perhaps a percent multiplier (e.g. +50%)... I mostly work with 80-column terminals, but sometimes make a terminal wide to be able to see wide output, but want to be able to quickly reset back to 80.

(You could argue the window manager/compositor should do this kind of thing, but I think it's on a grey edge, and the fact that a terminal shows a specifically sized grid and has these specific use cases means that I think it's in the best position to provide affordances of this nature. It's not like I want to be able to set any other window on my desktop to any particular width.)

With the appropriate APIs, this could be done in a kitten. (Maybe the APIs exist and I'm unaware of them? I'm having trouble finding a comprehensive reference to the APIs available in kittens.)

For extra credit, the ability to configure a subtle but visible "guide line" (like many editors have) at the 80c mark (or wherever the user wants) would be extra swell.

**Comment by kovidgoyal** [comment]
See #3788

**Reply by egnor** [reply]
Oooh :tada:
